### DISCUSSION TITLE
[bug?] Bypassing validations on update

### DISCUSSION THREAD

**Discussion by zbrunoneves** [mongodb/mongoid#5735] [Q&A] [answered]
Let's say I have an application with a `has_and_belongs_to_many` relation between a `Client` model and a `Company` model. 
In the `Client` model there is a validation to make sure the object is associated with at least one company (`companies` cannot be blank).
The thing is, when I try to update a `client` with an empty list of companies, despite raising a validation error message, the invalid model is updated in the database.
Is this an expected behavior?

One can reproduce it by running the following code:
```ruby
class Company
  include Mongoid::Document

  field :name, type: String
end

class Client
  include Mongoid::Document

  field :name, type: String

  has_and_belongs_to_many :companies, class_name: 'Company'

  validates :companies, presence: true
end

c = Client.first
c.update!(companies: []) # raises error but updates document
```

**Comment by alexbevi** [comment, marked as answer]
Hi @zbrunoneves, and thanks for reporting this! I've filed [MONGOID-5694](https://jira.mongodb.org/browse/MONGOID-5694) to have the team review this and identify next steps.

**Comment by roloand** [comment]
@alexbevi Sorry if this isn't according to protocol, I wanted to report, as I've done some testing while trying to upgrade Mongoid, this issue affects 7.3.5, 7.4.x, and 7.5.x

**Reply by alexbevi** [reply]
Thanks for the report @roloand - I've updated the _Affects Version/s_ on MONGOID-5694 accordingly. This ticket is still on our radar but won't be picked up until after we release Mongoid 9.0 (which should be happening soon).
